app/src/main/java/org/thoughtcrime/securesms/database/MmsSmsDatabase.java

```diff
@@ -295,24 +295,19 @@ public Set<Long> getAllMessageIdsFromSenderInThread(long threadId, String serial
     return identifiedMessages;
   }
 
-  public long getLastSentMessageFromSender(long threadId, String serializedAuthor) {
-
-    // Early exit
-    boolean isOwnNumber = Util.isOwnNumber(context, serializedAuthor);
-    if (!isOwnNumber) {
-      Log.i(TAG, "Asked to find last sent message but sender isn't us - returning null.");
-      return -1;
-    }
-
+  public long getLastOutgoingMessage(long threadId) {
     String order = MmsSmsColumns.NORMALIZED_DATE_SENT + " DESC";
     String selection = MmsSmsColumns.THREAD_ID + " = " + threadId;
 
-    // Try everything with resources so that they auto-close on end of scope
+    // Try everything with resources so that they auto-close on end of scope.
+    // Note: Do NOT call cursor.moveToFirst() once we have it, for reasons unknown to me it breaks the functionality. -AL
```

Comment:
It's because cursor starts at `-1` and `readerFor` calls `moveToNext`, which moves to `0` before returning the item. So you'd miss an item. It looks like this pattern is used 30 times in the app, so probably not worth commenting here.

It's awkward to comment on this because it's kind of just a pattern that's used in all of the databases, but yeah, if you think a comment adds clarification it'd probably make more sense as javadoc on the method, as

Reply:
Thanks for the info! Now that I know that I'll just remove the comment.